### PR TITLE
[MBL-871] Add compose visuals based on design system

### PR DESCRIPTION
**Pull request by mtgriego**
# 📲 What

Added many default views to be used in the future, added a screen where you can preview all the default views and typography 

# 🤔 Why

We need to prepare for our future visuals building

# 🛠 How

Compose and a bit of learning

# 👀 See

In App on internal builds: Go to internal tools and design system

# Story 📖

[MBL-871](https://kickstarter.atlassian.net/browse/MBL-871)

[MBL-871]: https://kickstarter.atlassian.net/browse/MBL-871?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/extensions/SafeLet.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.kickstarter.ui.extensions
+
+inline fun <T1 : Any, T2 : Any, R : Any> safeLet(p1: T1?, p2: T2?, block: (T1, T2) -> R?): R? {
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/extensions/SafeLet.kt:3]: I love this! but a question, could we user varargs to reduce the amount of `safeLet` declarations?

**Comment by Arkariang** [comment]
We could probably add to `jacoco.gradle` any new or modified folder containing only UI, that or create a spike ticket to explore UI testing with compose. The only UI testing we currently have it's screenshot testing and it's not adapted to compose yet.

**Comment by Arkariang** [comment]
Small comments but overall looks fantastic!

**Review by Arkariang — DISMISSED** [review]

File: app/src/main/java/com/kickstarter/ui/compose/designsystem/KSControls.kt [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+            checkmarkColor = colors.kds_white,
+            disabledColor = kds_support_500,
+        ),
+        enabled = enabled
+    )
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun KSStringDropdown(
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/compose/designsystem/KSControls.kt:159]: A bit worried about the `ExperimentalMaterialApi` for dropdowns, but one of the first UI pieces I would like to kill and substitute over this component will be the Shipping rules dropdown. The usability will dramatically improve.

File: app/src/main/java/com/kickstarter/ui/extensions/SafeLet.kt (hunk 1 above)
Comment (mtgriego) [reply app/src/main/java/com/kickstarter/ui/extensions/SafeLet.kt:3]: I am not sure, but I'll see what I can find!

File: app/src/main/java/com/kickstarter/ui/compose/designsystem/KSControls.kt (hunk 2 above)
Comment (mtgriego) [reply app/src/main/java/com/kickstarter/ui/compose/designsystem/KSControls.kt:159]: Yeah this was the only one that needed it fortunately.  The non experimental one didn't seem to match what we needed so I opted for this, but if something better replaces it all the better

File: app/src/main/java/com/kickstarter/ui/extensions/SafeLet.kt (hunk 1 above)
Comment (mtgriego) [reply app/src/main/java/com/kickstarter/ui/extensions/SafeLet.kt:3]: from what i can tell I can do a vararg, but the problem is that seems to make it so all the elements need to be the same type, whereas this allows each thing being passed to be whatever it wants to be as long as its not null.  ill keep digging on that when i can, but for now I will push this up

**Review by Arkariang — DISMISSED** [review]

**Review by Arkariang — APPROVED** [review]

**Pull request merged**